Tasks/Common/SonarQubeHelpers/SonarQubeHelper.ps1

```diff
@@ -202,16 +201,21 @@ function Assert
     }
 }
 
-
 #
-# Returns true if this build was triggered in response to a PR 
+# Returns true if this build was triggered in response to a PR
+#
+# Remark: this logic is temporary until the platform provides a more robust way of deteriming PR builds
 #
 function IsPrBuild
 {
-    $prcaEnabledString = GetTaskContextVariable "PullRequestSonarQubeCodeAnalysisEnabled"
-    $isPrBuild = Convert-String $prcaEnabledString Boolean
+    $sourceBranch = $env:Build_SourceBranch
 
-    return $isPrBuild;
+    if ([string]::IsNullOrWhiteSpace($sourceBranch))
+    {
+        throw "Could not determine the source branch, please log a bug"
+    }
+    
+    return $sourceBranch.StartsWith("refs/pull/", [StringComparison]::OrdinalIgnoreCase)
```

Comment:
A very good point. PRs are only relevant on TfsGit at this point. I am adding an extra check for the Repository Provider.